src/deployments/cdk/src/common/iam-assets.ts

```diff
@@ -109,13 +109,15 @@ export class IamAssets extends cdk.Construct {
       trustPolicy?: string,
     ): iam.Role => {
       let newRole: iam.Role | undefined;
-      if (sourceAccount && sourceAccountRole) {
+      if (sourceAccount) {
         const sourceAccountId = getAccountId(accounts, sourceAccount);
 
         newRole = new iam.Role(this, `IAM-Role-${role}-${accountKey}`, {
           roleName: role,
           description: `PBMM - ${role}`,
-          assumedBy: new iam.ArnPrincipal(`arn:aws:iam::${sourceAccountId}:role/${sourceAccountRole}`),
+          assumedBy: sourceAccountRole
+            ? new iam.ArnPrincipal(`arn:aws:iam::${sourceAccountId}:role/${sourceAccountRole}`)
+            : new iam.ArnPrincipal(`arn:aws:iam::${sourceAccountId}:root`),
```

Comment:
Dislike this change, this opens all roles up to being assumed by any user in the account.  If you want to add root to the assumption policy, please use the trust policy (which was fixed in v1.5) - see [sample snippets](https://github.com/aws-samples/aws-secure-environment-accelerator/blob/main/reference-artifacts/SAMPLE_CONFIGS/sample_snippets.md) "Cross Account Role Example with trust policies (trust policies not supported until v1.5.0+)"

Reply:
This change has been reverted. The OpenSearch admin role, that is needed on creation, is now the execution role for the Lambda that configures OpenSearch. The role that was created in the configuration used by this change has also been removed.